corehq/apps/reports/generic.py

```diff
@@ -437,7 +437,7 @@ def _update_initial_context(self):
 
         def is_datespan(field):
             field_fn = to_function(field) if isinstance(field, basestring) else field
-            return issubclass(field_fn, DatespanFilter)
+            return issubclass(field_fn, DatespanFilter) and field_fn.is_viewable
```

Comment:
yeah thanks, that makes it much clearer. any reason not to make those name changes if you haven't already?